lib/fabrik/VirtualHostAgent.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Agent = require('./Agent');
+
+class VirtualHostAgent extends Agent {
+
+  constructor(settings) {
+    super({
+      json: true
+    });
+    this.settings = settings;
+  }
+
+  provision(ips, guid) {
+    const body = {
+      parameters: {}
+    };
+    return this
+      .getHost(ips, 'multi-tenancy')
+      .then(ip => this.post(ip, `tenants/${guid}/lifecycle/provision`, body, 200));
+  }
+
+  deprovision(ips, guid) {
+    const body = {
+      parameters: {}
+    };
+    return this
+      .getHost(ips, 'multi-tenancy')
+      .then(ip => this.post(ip, `tenants/${guid}/lifecycle/deprovision`, body, 200));
```

Comment:
wrong HTTP semantics. This should be HTTP Delete on the resource `tentants/$guid`